### PR TITLE
Actually make auth tokens longer (40 chars)

### LINKED ISSUE #2724: Make the auth token LONGER
This is a breaking change, as it would invalidate all current auth token url.

But this change is needed.

Current implementation uses 8 chars: I propose 40 (SHA 1 = 160 bits, not truncated, which means 40 characters in HEX)

I propose to use sha256 also.

Re #2711

### PR DESCRIPTION
**Pull request by nitriques**
Fixes #2724

Both reset password token and auth token are 40 chars long.
Database is changed to take 255 long token (for the future)

Auth token are generated on demand. Deactivating the remote login deletes the token. Renabling it creates a new token.

### CODE REVIEW

File: symphony/lib/toolkit/class.authormanager.php [hunk 1]
```diff
@@ -81,6 +81,43 @@ public static function delete($id)
             ->success();
     }
 
+    /**
+     * Undocumented function
```
Comment (michael-e) [inline symphony/lib/toolkit/class.authormanager.php]: "Undocumented function"? Looks like an oversight to me.

Comment (michael-e) [reply symphony/lib/toolkit/class.authormanager.php]: Ah, you did it later!

Comment (nitriques) [reply symphony/lib/toolkit/class.authormanager.php]: ;)

**Comment by nitriques** [comment]
Yes @michael-e there are no hurry to merge it anyways. I thought it would be small.. but you know how it goes ;)

**Comment by michael-e** [comment]
Oh yes, I know. Features are like icebergs. You only see a small part…

File: symphony/lib/toolkit/class.cryptography.php [hunk 2]
```diff
@@ -96,4 +96,31 @@ public static function generateSalt($length)
...
+            throw new Exception('Can not generate less than 16 random bytes');
+        }
+        if (function_exists('random_bytes')) {
+            return bin2hex(random_bytes($length / 2));
+        } elseif (function_exists('openssl_random_pseudo_bytes')) {
+            return bin2hex(openssl_random_pseudo_bytes($length / 2));
+        }
+        return self::generateSalt($length);
+    }
 }
```
Comment (michael-e) [inline symphony/lib/toolkit/class.cryptography.php]: I like the way this function works. Checking `random_bytes` exists is a good idea.

**Comment by nitriques** [comment]
Thanks @michael-e !

**Pull request merged**